lib/matplotlib/backend_tools.py

```diff
@@ -1020,6 +1021,48 @@ def _mouse_move(self, event):
         self.toolmanager.canvas.draw_idle()
 
 
+class ToolHelpBase(ToolBase):
+    description = 'Print tool list, shortcuts and description'
+    default_keymap = rcParams['keymap.help']
+    image = 'help.png'
+
+    @staticmethod
+    def format_shortcut(key_sequence):
+        """
+        Converts a shortcut string from the notation used in rc config to the
+        standard notation for displaying shortcuts, e.g. 'ctrl+a' -> 'Ctrl+A'.
+        """
+        return (key_sequence if len(key_sequence) == 1 else
+                re.sub(r"\+[A-Z]", r"+Shift\g<0>", key_sequence).title())
```

Comment:
This doesn't add Shift for single letter shortcuts like 'L' for xscale; is that on purpose?

Reply:
Good question. I've been implicitly assuming that we keep the case sensitivity for single letters, i.e.
~~~
xscale k,L
yscale l
~~~

Now thinking about it, this is probably what we want to do. Otherwise, we'd have
~~~
xscale K, Shift+L
yscale L
~~~
This could be slightly misleading, because "L" really means only typing the L-Key, which in a text field would result in a lowercase "l". An alternative way out would be to enclose the keys.
~~~
xscale <K>, <Shift>+<L>
yscale <L>
~~~

Note: Shortcuts equal to single upper/lowercase letters are rarely used, because they clash with text input. I'm not aware of a standard notation for these cases.

Reply:
vim just uses lowercase for lowercase and uppercase for uppercase (http://vimdoc.sourceforge.net/htmldoc/editing.html#CTRL-^), which seems just fine to me.  ctrl+x are always documented as CTRL-X (UPPERCASE), but there are no shortcuts of the form ctrl+shift+x due to internal constraints.